pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -1037,6 +1039,10 @@ func k3sBasedClusterConfig(cluster *v3.Cluster, nodes []*v3.Node) error {
 			cluster.Spec.K3sConfig.SetStrategy(1, 1)
 		}
 	} else if strings.Contains(cluster.Status.Version.String(), "rke2") {
+		if cluster.Name == "local" {
```

Comment:
How can we tell if user just install rancher on rke2 cluster but not from rancherd? does the provider also become rancherd?

Reply:
oh right. hmph.

Reply:
Not sure if installing rancher on rke2 will be a use case, if so we need to tell the difference between rke2 cluster and a rancherd rke2 cluster. Right now we can say if it is rke2 then it is always going to be rancherd rke2.

Reply:
I would like to see this fixed in the initial release. As is, an rke2 cluster will look like rancherd and then if we fix it in a later release, the cluster's type would change from rancherd to rke2 upon upgrade, which may cause an issue.

Is there some simple thing we can look for in the cluster to "know" it was a rancherd2 install? Heck, if we just added a check for a daemonset named rancher, that would probably be a good enough indication. (This is because rancher is typically installed as a deployment, not a daemonset, but in the new chart that @StrongMonkey developed for rancherd, we changed that to a daemonset)

Reply:
:+1: